com.unity.multiplayer.mlapi/Runtime/Spawning/NetworkSpawnManager.cs

```diff
@@ -225,6 +225,8 @@ internal NetworkObject CreateLocalNetworkObject(bool softCreate, uint prefabHash
                     // Otherwise, instantiate an instance of the NetworkPrefab linked to the prefabHash
                     var networkObject = ((position == null && rotation == null) ? UnityEngine.Object.Instantiate(networkPrefabReference) : UnityEngine.Object.Instantiate(networkPrefabReference, position.GetValueOrDefault(Vector3.zero), rotation.GetValueOrDefault(Quaternion.identity))).GetComponent<NetworkObject>();
 
+                    networkObject.NetworkManagerTestOverride = NetworkManager;
```

Comment:
Probably worth a comment saying something like "by default all network objects are spawned with this singleton, central NM which can be overridden for testing"

Reply:
as per my comments in #762, I'm still not 100% onboard with this approach yet.

Reply:
since now we're aligned, I'm onboard with our short-term solution.